pandas/tests/indexes/multi/test_reshape.py

```diff
@@ -175,6 +175,6 @@ def test_delete_base(idx):
     assert result.equals(expected)
     assert result.name == expected.name
 
-    with pytest.raises((IndexError, ValueError)):
-        # Exception raised depends on NumPy version.
+    msg = "index 6 is out of bounds for axis 0 with size 6"
+    with pytest.raises(IndexError, match=msg):
         idx.delete(len(idx))
```

Comment:
In earlier versions NumPy used to [raise ValueError](https://github.com/numpy/numpy/blob/v1.2.0/numpy/lib/function_base.py#L2756), pandas [setup.py has defined min_numpy_version to 1.13.3](https://github.com/pandas-dev/pandas/blob/master/setup.py#L36). Since [NumPy 1.13.3, IndexError is being raised](https://github.com/numpy/numpy/blob/v1.13.3/numpy/lib/function_base.py#L4842)

Reply:
@sathyz Thank you for looking into this!